cloud/scope/cluster.go

```diff
@@ -165,6 +173,11 @@ func (s *ClusterScope) NodeSubnet() *infrav1.SubnetSpec {
 	return s.AzureCluster.Spec.NetworkSpec.GetNodeSubnet()
 }
 
+// RouteTable returns the cluster node routetable.
```

Comment:
How do you feel about also fixing https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/718 as part of this PR?

Reply:
I was going to fix that next, but will add it here as a 2nd commit

Reply:
there is a comment into the issue saying `I am not 100% sure it makes sense to have two different route tables. Need to look into it more.` 

do you think its valid to have two routetables?

Reply:
as discussed on slack, requirements aren't clear yet so let's merge this refactor as-is and open a separate PR to fix the bug which is pre-existing